elytron/src/main/java/org/wildfly/extension/elytron/FileSystemRealmDefinition.java

```diff
@@ -110,7 +122,24 @@
             .setAllowExpression(true)
             .build();
 
-    static final AttributeDefinition[] ATTRIBUTES = new AttributeDefinition[]{PATH, RELATIVE_TO, LEVELS, ENCODED, HASH_ENCODING, HASH_CHARSET};
+    static final SimpleAttributeDefinition CREDENTIAL_STORE =
+            new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.CREDENTIAL_STORE, ModelType.STRING, false)
+                    .setAllowExpression(true)
+                    .setRequired(false)
```

Comment:
It should be ensured the attributes require each other.

Reply:
Does configuring just CREDENTIAL_STORE without SECRET_KEY makes sense? Is the first or the only key used? What in case there are multiple keys?

Reply:
Fixed to both being required. `CREDENTIAL_STORE` requires `SECRET_KEY` and `SECRET_KEY` requires `CREDENTIAL_STORE`